### PR TITLE
azure v5: modify node pool

### PR DESCRIPTION
**Pull request by axbarsan**
Towards https://github.com/giantswarm/giantswarm/issues/12535

This PR:
* Supports changing a node pool name (scaling is unsupported at this moment)

### Preview

<details>
<summary>Command help</summary>

image

</details>

<details>
<summary>Update name</summary>

image

</details>

<details>
<summary>Trying to use autoscaling (unsupported right now)</summary>

image

</details>

<details>
<summary>Trying to update... nothing</summary>

image

</details>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -15,7 +15,7 @@ require (
 	github.com/giantswarm/gsclientgen v1.0.2-0.20200724110725-6bd63fe7f9c6
 	github.com/giantswarm/k8sclient v0.0.0-20191213144452-f75fead2ae06
 	github.com/giantswarm/kubeconfig v0.0.0-20191209121754-c5784ae65a49
-	github.com/giantswarm/microerror v0.2.0
+	github.com/giantswarm/microerror v0.2.1
```
Comment (axbarsan) [inline go.mod:18]: Updating to be able to use `microerror.Pretty()`

File: commands/update/nodepool/command.go [hunk 2]
```diff
@@ -113,10 +130,23 @@ func verifyPreconditions(args Arguments) error {
...
+		if args.ScalingMin == 0 && args.ScalingMax == 0 && args.Name == "" {
+			return microerror.Maskf(errors.NoOpError, "Nothing to update.")
+		} else if args.ScalingMin > args.ScalingMax && args.ScalingMax > 0 {
+			return microerror.Mask(errors.WorkersMinMaxInvalidError)
+		}
+	}
+
+	if args.Provider == provider.Azure {
+		if args.ScalingMin > 0 || args.ScalingMax > 0 {
+			return microerror.Maskf(errors.NoOpError, "Provider '%s' does not support node pool scaling.", args.Provider)
```
Comment (tuommaki) [inline commands/update/nodepool/command.go:145]: Azure does support node pool size change, but not auto-scaling. Should the check maybe be `args.ScalingMin != args.ScalingMax` ?

Comment (axbarsan) [reply commands/update/nodepool/command.go]: I updated the code and the screenshots 👍

File: commands/update/nodepool/command.go [hunk 3]
```diff
@@ -113,10 +132,22 @@ func verifyPreconditions(args Arguments) error {
...
+
+	switch args.Provider {
+	case provider.AWS:
+		if args.ScalingMin > args.ScalingMax && args.ScalingMax > 0 {
+			return microerror.Mask(errors.WorkersMinMaxInvalidError)
+		}
+
+	case provider.Azure:
+		if args.ScalingMin != args.ScalingMax {
+			return microerror.Maskf(errors.NoOpError, "Provider '%s' does not support node pool autoscaling.", args.Provider)
```
Comment (tuommaki) [inline commands/update/nodepool/command.go:149]: Isn't `WorkersMinMaxInvalidError` for validation error?

```suggestion
			return microerror.Maskf(errors.WorkersMinMaxInvalidError, "Provider '%s' does not support node pool autoscaling.", args.Provider)
```

Comment (axbarsan) [reply commands/update/nodepool/command.go]: Nice one! Thank you 👍

**Pull request merged**